.github/workflows/ci.yml

```diff
@@ -0,0 +1,80 @@
+# SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+name: CI Build
+
+# Controls when the workflow will run.
+on:
+  # run pipeline on push event of main branch.
+  push:
+    branches:
+      - main
+  # run pipeline on pull request
+  pull_request:
+  # run pipeline on merge queue
+  merge_group:
+  # run this workflow manually from the Actions tab
+  workflow_dispatch:
+  # Nightly run
+  schedule:
+    - cron: "0 2 * * *" # Based on UTC time
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-ci-build
+  cancel-in-progress: true
+
+jobs:
+  build-test-release:
+    name: build-test-release
+    uses: "./.github/workflows/build-test-release.yml"
+    permissions:
+      contents: write
+    with:
+      create_release: false
+
+  check-code-quality:
+    uses: "./.github/workflows/check-code-quality.yml"
+
+  reuse-compliance:
+    uses: "./.github/workflows/reuse-compliance.yml"
+
+  clang-tidy:
+    uses: "./.github/workflows/clang-tidy.yml"
+    with:
+      target: "all power_grid_model_benchmark_cpp"
+
+  check-citations:
+    runs-on: ubuntu-24.04
+    # we need to define the output to access the output of the current job in the next job 
+    outputs:
+      citation_modified: ${{ steps.check.outputs.citation_modified }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        # checkout the last commit
+        with:
+          fetch-depth: 2
+
+      - name: Get changed files
+        run: |
+          CHANGED_FILES=$(git diff --name-only HEAD^1 HEAD | tr '\n' ' ')
+          echo "$CHANGED_FILES"
+          echo "changed_files=$CHANGED_FILES" >> $GITHUB_ENV
+
+      - name: Check if CITATION.cff was modified
+        id: check
+        if: contains(env.changed_files, 'CITATION.cff') || contains(env.changed_files, '.github/workflows/citations.yml')
+        run: echo "citation_modified=True" >> $GITHUB_OUTPUT
+
+  citations:
+    needs: check-citations
+    if: needs.check-citations.outputs.citation_modified == 'True'
+    uses: "./.github/workflows/citations.yml"
```

Comment:
this looks more complex than it probably should be. i'm inclined to keep it a separate workflow for CI and only in the Nightly call it from here (like it was before)